extension/core_functions/scalar/struct/struct_values.cpp

```diff
@@ -0,0 +1,69 @@
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/execution/expression_executor_state.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp" // VariableReturnBindData
+#include "core_functions/scalar/struct_functions.hpp"
+
+namespace duckdb {
+
+static void StructValuesFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto &input = args.data[0];
+	const idx_t count = args.size();
+
+	auto &input_children = StructVector::GetEntries(input);
+	auto &result_children = StructVector::GetEntries(result);
+	D_ASSERT(result_children.size() == input_children.size());
+
+	// We would use result.Reference(input) here, but that function asserts that the logical types are the same
+	// UnnamedStruct vector and Struct vector are actually the same underneath, so we can just reference the children
+	for (idx_t i = 0; i < input_children.size(); i++) {
+		result_children[i]->Reference(*input_children[i]);
+	}
+
+	if (input.GetVectorType() == VectorType::CONSTANT_VECTOR) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		const bool is_null = ConstantVector::IsNull(input);
+		ConstantVector::SetNull(result, is_null);
+	} else {
+		result.SetVectorType(VectorType::FLAT_VECTOR);
+
+		// Make result validity to mirror input's nulls
+		UnifiedVectorFormat input_data;
+		input.ToUnifiedFormat(count, input_data);
+		auto &validity = FlatVector::Validity(result);
+
+		validity.EnsureWritable();
```

Comment:
I think we don't need this, `SetInvalid` will ensure this already

In the scenario where the input rows turn out to all be valid, if we use this, `AllValid()` will not be true for the result, because it checks whether EnsureWritable was ever called (explicitly or through methods like `SetInvalid`)

So we would be missing out on fast paths in downstream function calls